### PR TITLE
Add azure dev spaces

### PR DESCRIPTION
**Pull request by LeonidLapshin**
# Description
Add Azure Dev Spaces CLI to Windows 2016/2019 images

#### Related issue:
https://github.com/actions/virtual-environments/issues/1452

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1 [hunk 1]
```diff
@@ -74,6 +74,7 @@ $markdown += New-MDHeader "Tools" -Level 3
 $markdown += New-MDList -Style Unordered -Lines @(
     (Get-AzCosmosDBEmulatorVersion),
     (Get-AzCopyVersion),
+    (Get-AZDSVersion),
```
Comment (al-cheb) [inline images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1:77]: Why is the "Tools" section when you put tests in "CLI.Tools" ?

File: images/win/scripts/Installers/Install-AzureDevSpacesCLI.ps1 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+################################################################################
+##  File:  Install-AzureDevSpacesCLI.ps1
+##  Desc:  Install Azure Dev Spaces CLI
+################################################################################
+
+# Install Azure Dev Spaces CLI
+Install-Binary -Url "https://aka.ms/get-azds-windows" -Name "Azure Dev Spaces CLI.exe" -ArgumentList ("/quiet")
+$env:Path = $env:Path + ";C:\Program Files\Microsoft SDKs\Azure\Azure Dev Spaces CLI"
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-AzureDevSpacesCLI.ps1:8]: Please pay attention that this line will update Path only for this single script. If we need to add this folder to PATH permanently, we should use https://github.com/actions/virtual-environments/blob/main/images/win/scripts/ImageHelpers/PathHelpers.ps1#L30

File: images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1 (hunk 1 above)
Comment (maxim-lobanov) [inline images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1:77]: Let's move it to `CLI Tools` section in docs

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 [hunk 3]
```diff
@@ -246,4 +246,9 @@ function Get-VisualCPPComponents {
             }
         }
     }
-}
\ No newline at end of file
+}
+
+function Get-AZDSVersion {
+    $azdsVersion = $(azds --version) | select-string "(\d+\.\d+\.\d+.\d+)"
```
Comment (maxim-lobanov) [inline images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:252]: Please use PS notation for build-in functions: `Select-String`

**Comment by maxim-lobanov** [comment]
@LeonidLapshin how was it tested?
Have you validated that this step works after these changes?
image

**Review by maxim-lobanov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Comment by LeonidLapshin** [comment]
Azure Dev Spaces CLI successfully added to Windows 2016/2019 images and will be available with the next rollout (1-2 weeks) :)

**Pull request merged**